sdk/core/core/inc/az_span_reader.h

```diff
@@ -34,6 +36,25 @@ AZ_INLINE void az_span_reader_next(az_span_reader * const p_reader) {
   p_reader->i += 1;
 }
 
+// Parsing utilities
+
+AZ_NODISCARD AZ_INLINE az_result az_error_unexpected_char(az_result_byte const c) {
+  return az_failed(c) ? c : AZ_ERROR_UNEXPECTED_CHAR;
```

Comment:
I still really don't like the `az_result_foo` thing

Reply:
https://github.com/Azure/azure-sdk-for-c/issues/130